dejacode_toolkit/vex.py

```diff
@@ -0,0 +1,290 @@
+    "RB": ImpactAnalysisResponse.ROLLBACK,
+    "U": ImpactAnalysisResponse.UPDATE,
+    "WNF": ImpactAnalysisResponse.WILL_NOT_FIX,
+    "WA": ImpactAnalysisResponse.WORKAROUND_AVAILABLE,
+}
+
+
+def create_auto_vex(package, vulnerabilities):
+    # automatically create a VEX for each product package that has a vulnerability
+    vex_objects = []
+    vulnerability_ids = []
+    for entry in vulnerabilities:
+        unresolved = entry.get("affected_by_vulnerabilities", [])
+        for vulnerability in unresolved:
+            vulnerability_id = vulnerability.get("vulnerability_id")
+            if vulnerability_id:
+                vulnerability_ids.append(vulnerability_id)
+
+    productpackages = ProductPackage.objects.filter(package=package)
+    for productpackage in productpackages:
+        for vulnerability_id in vulnerability_ids:
+            vex_objects.append(
+                ProductPackageVEX(
+                    dataspace=productpackage.dataspace,
+                    productpackage=productpackage,
+                    vulnerability_id=vulnerability_id,
+                )
+            )
+
+    ProductPackageVEX.objects.bulk_create(vex_objects, ignore_conflicts=True)
+
+
+def vulnerability_format_vcic_to_cyclonedx(vcio_vulnerability, vex: ProductPackageVEX):
+    """Change the VCIO format of the vulnerability to CycloneDX and add the vex vulnerability"""
+    vulnerability_source = VulnerabilitySource(
+        url=vcio_vulnerability.get("url"),
+    )
+
+    references = vcio_vulnerability.get("references") or []
+    vulnerability_reference, vulnerability_ratings = get_references_and_rating(references)
+
+    state = STATE_VEX_CYCLONEDX.get(vex.state)
+    justification = JUSTIFICATION_VEX_CYCLONEDX.get(vex.justification)
+
+    responses = []
+    for vex_res in vex.responses or []:
+        response = RESPONSES_VEX_CYCLONEDX.get(vex_res)
+        if response:
+            responses.append(response)
+
+    vulnerability_analysis = VulnerabilityAnalysis(
+        state=state,
+        responses=responses,
+        justification=justification,
+        detail=vex.detail,
+    )
+
+    # vulnerability_advisory = VulnerabilityAdvisory()  # ignore
+    # vulnerability_credits = VulnerabilityCredits()  # ignore
+
+    # property = Property()  # ignore
+    # tool = Tool()  # ignore
+
+    bom_targets = []
+    versions = []
+    for affected_package in vcio_vulnerability.get("affected_packages") or []:
+        is_vulnerable = affected_package.get("is_vulnerable")
+        status = STATUS_VEX_CYCLONEDX.get(is_vulnerable, ImpactAnalysisAffectedStatus.UNKNOWN)
+        purl_string = affected_package.get("purl")
+
+        vul_purl = None
+        if purl_string:
+            vul_purl = PackageURL.from_string(purl_string)
+
+        versions.append(BomTargetVersionRange(version=vul_purl.version, status=status))
+
+    if versions:
+        bom_target = BomTarget(ref="urn:cdx:serialNumber/version#bom-ref", versions=versions)
```

Comment:
I am confused about [bom_target](https://github.com/CycloneDX/bom-examples/blob/master/VEX/vex.json#L91)
especially this part `urn:cdx:3e671687-395b-41f5-a30f-a58921a69b79/1#`
where can we get bom ref? ( vulnerablecode doesn't provide the bom ref  ) should be create a bom and pass the bom ref to vex